File: JavaServer/Qipai4Core/src/net/silverfoxserver/core/service/Mail.java
/*
 * SilverFoxServer: massive multiplayer game server for Flash, ...
 * VERSION:3.0
 * PUBLISH DATE:2015-9-2 
 * GITHUB:github.com/wdmir/521266750_qq_com
 * UPDATES AND DOCUMENTATION AT: http://www.silverfoxserver.net
 * COPYRIGHT 2009-2015 SilverFoxServer.NET. All rights reserved.
 * MAIL:521266750@qq.com
 */
package net.silverfoxserver.core.service;

import net.silverfoxserver.core.model.IUserModel;
import java.time.LocalDate;
import java.time.LocalTime;

public class Mail
{
    /** 
     收信人
    */
    public IUserModel fromUser;

    /** 
     收件人
    */
    public IUserModel toUser;

    /** 

    */
    public String n;

    public String param;

    /** 
     过期,根据时间来
     上层逻辑根据过期几天来删除没发送成功的mail
    */
    public int dayOfYear;

    public Mail(IUserModel fromUser, IUserModel toUser, String n, String param)
    {
            //
            this.fromUser = fromUser;
            this.toUser = toUser;
            this.n = n;
            this.param = param;

            //
            this.dayOfYear = LocalDate.now().getDayOfYear();//DayOfYear;
    }

    /** 


     @return 
    */
    public final String toXMLString()
    {
            StringBuilder sb = new StringBuilder();

            //f = from
            //t = to
            //p = param
            sb.append("<mail n='" + this.n + "'>");
            sb.append("<f>");
            sb.append(fromUser.toXMLString());
            sb.append("</f>");
            sb.append("<t>");
            sb.append(toUser.toXMLString());
            sb.append("</t>");
            sb.append("<p>");
            sb.append(this.param);
            sb.append("</p>");
            sb.append("</mail>");

            return sb.toString();

    }

}

File: CSharpServer/Qipai4Core/Qipai4Core/net/silverfoxserver/core/service/Mail.cs
/*
* SilverFoxServer: massive multiplayer game server for Flash, ...
* VERSION:3.0
* PUBLISH DATE:2015-9-2 
* GITHUB:github.com/wdmir/521266750_qq_com
* UPDATES AND DOCUMENTATION AT: http://www.silverfoxserver.net
* COPYRIGHT 2009-2015 SilverFoxServer.NET. All rights reserved.
* MAIL:521266750@qq.com
*/
using System;
using System.Collections.Generic;
using System.Text;
using net.silverfoxserver.core.model;

namespace net.silverfoxserver.core.service
{
    public class Mail
    {
        /// <summary>
        /// 收信人
        /// </summary>
        public IUserModel fromUser;

        /// <summary>
        /// 收件人
        /// </summary>
        public IUserModel toUser;

        /// <summary>
        /// 
        /// </summary>
        public string n;

        public string param;

        /// <summary>
        /// 过期,根据时间来
        /// 上层逻辑根据过期几天来删除没发送成功的mail
        /// </summary>
        public int dayOfYear;

        public Mail(IUserModel fromUser, IUserModel toUser, string n,string param)
        {
            //
            this.fromUser = fromUser;
            this.toUser = toUser;
            this.n = n;
            this.param = param;

            //
            this.dayOfYear = DateTime.Now.DayOfYear;
        }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public string toXMLString()
        {
            StringBuilder sb = new StringBuilder();

            //f = from
            //t = to
            //p = param
            sb.Append("<mail n='" + this.n + "'>");
            sb.Append("<f>");
            sb.Append(fromUser.toXMLString());
            sb.Append("</f>");
            sb.Append("<t>");
            sb.Append(toUser.toXMLString());
            sb.Append("</t>");
            sb.Append("<p>");
            sb.Append(this.param);
            sb.Append("</p>");
            sb.Append("</mail>");

            return sb.ToString();
        
        }

    }
}
